### ISSUE TITLE
Opprette NamingSystem - FEST-id

### ISSUE THREAD

**Issue by rockphotog** [HL7Norway/basisprofiler-r4#16]
Sjekk i M30 FEST-dokumentasjonen om det er snakk om 1 eller to konsept. 

Inspirasjon:
https://simplifier.net/HL7Norwayno-basis/2.16.578.1.12.4.1.4.4/

**Comment by rockphotog** [comment]
Ligger i egen branch:
https://github.com/HL7Norway/basisprofiler-r4/tree/namingsystem-fest-id/NamingSystem

**Comment by oaassv** [comment]
@stinejohans1 
Det er vel naturlig at NamingSystem for Fest-ID knyttes til SLV, på samme måte som kommende profiler for Medication, Substance ++. I en drømmeverden hadde disse vært knyttet til en URL som ender opp med å bli resolvable i nye SAFEST, men der er vi vel ikke helt enda. Uansett, hvis dere har et forslag til URL for identifikasjon av ressurser for legemiddelområdet tar vi gjerne imot forslag ex www.slf.no/fhir/fest-id.

**Comment by kennethmyhra** [comment]
Burde være http://slf.no/fhir/NamingSystem/fest-id slik at den kan være resolvable.

**Comment by rockphotog** [comment]
Syklistenes landsforening, der er jeg medlem :-)

Bruker dette inntil videre:

```
<uniqueId>
      <type value="uri" />
      <value value="http://legemiddelverket.no/fhir/NamingSystem/fest-id" />
      <preferred value="true" />
</uniqueId>
```

**Comment by stinejohans1** [comment]
Nå vet ikke jeg om ID'ene i FEST er unike på tvers av kataloger og typer, men burde det være et namingsystem per type? 

Eksempelvis i profilen for MedicinalProductPackaged er det kun ID'en LegemiddelPakningID i LegemiddelPakningMerkevare-katalogen som er relevant å bruke.

**Comment by rockphotog** [comment]
Det kan vi finne ut. Mente de var basert på UUID slik at de var unike uavhengig av katalog. IDMP har vel en tre-fire forskjellige ID'er, som MPID, PhID, etc.

**Comment by rockphotog** [comment]
@olemartinwinnem Stemmer det et FEST-id'ene er unike, uavhengig av katalog?

**Comment by thomiz** [comment]
@rockphotog @stinejohans1 @olemartinwinnem 
Jeg tror ikke vi skal ha FEST-id som NamingSystem, så lenge vi betrakter FEST-id som kodeverk og ikke en identifikator.

**Comment by rockphotog** [comment]
Kan et CodeSystem være "tomt" (uten faktiske koder), på samme måte som NamingSystem? Jeg tror aldri det vil finnes en kuratert liste over gyldige FEST-id, det er nok antageligvis for mye bruk og kast.

**Comment by AndersAlmas** [comment]
I følge spec. en på code system så er concept-attributtet (det som angir innholdet) 0..* kardinalitet da. Så det tyder på at en kan ha et "tomt" codesystem.

**Comment by rockphotog** [comment]
Bra. Så sånn sett spiller det ikke noen stor rolle om FEST-id er definert i et CodeSystem eller NamingSystem, det viktigste er av vi har den kontrollert system-id, type `http://legemiddelverket.no/fhir/DEBATTsystem/fest-id` for bruk i coding.system.

**Comment by thomiz** [comment]
@rockphotog Vil ikke alle FEST-id'er som skal brukes i helsetjenesten ligge i FEST uttrekk? Jeg trodde det var der de ble kuratert og meningen med dem ble definert.

**Comment by thomiz** [comment]
@rockphotog Vi har allerede et kontrollert system-id definert i no-basis, så hva er problemet?
http://ehelse.no/fhir/CodeSystem/FEST

**Comment by rockphotog** [comment]
Ikke noe problem, utover at jeg mener FEST-id ikke er et kodeverk (men et "open ended sett av databasenøkler"...). Funksjonen er det samme.

**Comment by thomiz** [comment]
De fleste større kodeverk er i praksis open-ended med koder som ikke har noen sammenheng med definisjonen av konseptene, så overgangen mellom kodeverk og namingsystem vil alltid være flytende i disse tilfellene. FHIR spekken sier at naming system kan benyttes både til å navngi identifikatorer og kodeverk, mens vi i norge har tenkt å benytte namingsystem hovedsaklig for å identifisere identifikator-typer/system. 

I praksis er det vel heller ingen entydig definisjon av forskjellen på en kode og en identifikator selv om ISO har prøvd seg: 
*ISO/IEC 11179 metadata registry standard defines a code as system of valid symbols that substitute for longer values *

Mens ISO kodeverket for kjønn ser slik ut: 
https://en.wikipedia.org/wiki/ISO/IEC_5218

Vi har ingen entydig logikk på Volvenkodeverk om at kodene skal/ikke skal ha noen semantisk verdi.

Verdier fra SNOMED CT terminologien er jo helt vanlige identifikatorer uten mening, men ansees som et kodeverk i FHIR sammenheng:
https://www.hl7.org/fhir/valueset-approach-site-codes.html
https://www.hl7.org/fhir/terminologies-systems.html

Vi får uansett problemer med validering av kodene i disse store kodeverkene, så slik sett hadde det vært enklere for oss å anse dem som identifikatorer. Når vi skal validere instanser av no-bais-Medication ressursen, som bruker koder fra http://ehelse.no/fhir/CodeSystem/FEST så vil ihvertfall den offisielle validatoren forsøke å finne kodeverket i form av en CodeSystem instans. Hvis den ikke finner CodeSystem instansen krasjer validatoren men hvis CodeSystem instansen den finner ikke inneholder koder vil den kaste en feil som sier at koden ikke eksisterer i kodeverket og derfor har Medication instansen feil. Jeg kjenner ikke til noen måte å unngå akkurat det på, unntatt å faktisk finne koden i FEST da selvfølgelig.

**Comment by rockphotog** [comment]
Koder bør uansett ikke ha noen semantisk verdi, men det gir jo god lesbarhet. 

For ATC er "C02A C05" en kode for "Moksonidin", for ICD-10 er "R12" kode for "Halsbrann". 
For FEST-id er det en katalogkode som indirekte peker på diverse konsepter i den underliggende strukturen, som "navn, form, styrke": Kun en identifikator. 

Man _kunne_ sagt at koden "ID_3FBC70B6-62EF-4F03-9C13-D0FA701200C3" = "Minirin Tab 0,1 mg", men det er jo ikke tilfelle. Det er mulig at det ligger slik internt hos SLV, men dette vil uansett IDMP løse mye bedre med PhID, MPID, etc. 

```xml
<OppfLegemiddelMerkevare>
      <Id>ID_3FBC70B6-62EF-4F03-9C13-D0FA701200C3</Id>
      <Tidspunkt>2019-02-13T18:20:15</Tidspunkt>
      <Status V="A" DN="Aktiv oppføring" />
      <LegemiddelMerkevare xmlns="http://www.kith.no/xmlstds/eresept/forskrivning/2014-12-01">
        <Atc V="H01BA02" S="2.16.578.1.12.4.1.1.7180" DN="Desmopressin" />
        <NavnFormStyrke>Minirin Tab 0,1 mg</NavnFormStyrke>
        <Reseptgruppe V="C" DN="Reseptgruppe C" />
        <LegemiddelformKort V="53" S="2.16.578.1.12.4.1.1.7448" DN="Tablett" />
        <RefVilkar>ID_491E7F6C-23A9-45C3-A906-95A1332AD5BD</RefVilkar>
        <Preparattype V="11" DN="Krever godkj. Fritak" />
        <TypeSoknadSlv V="2" DN="Søknad vurderes av apotek" />
        <AdministreringLegemiddel>
          <Administrasjonsvei V="53" S="2.16.578.1.12.4.1.1.7477" DN="Oral bruk" />
          <EnhetDosering V="13" S="2.16.578.1.12.4.1.1.7480" DN="tablett" />
        </AdministreringLegemiddel>
        <Id>ID_000C0AFE-42AE-4522-AE22-BAAF5746D81E</Id>
        <Varenavn>Minirin</Varenavn>
        <LegemiddelformLang>Tablett</LegemiddelformLang>
        <SortertVirkestoffMedStyrke>
          <Sortering>0</Sortering>
          <RefVirkestoffMedStyrke>ID_9006F019-8E46-4003-BA2B-2C41DACF8CC1</RefVirkestoffMedStyrke>
        </SortertVirkestoffMedStyrke>
        <ProduktInfo>
          <Produsent>Ferring</Produsent>
        </ProduktInfo>
        <Reseptgyldighet>
          <Varighet>P1Y</Varighet>
        </Reseptgyldighet>
      </LegemiddelMerkevare>
    </OppfLegemiddelMerkevare>
```

Ok, slutt på rant at eResept har brukt en database*nøkkel* (ikke ID) for å rekvirere medisiner :-) 
CodeSystem it is!

> Jeg kjenner ikke til noen måte å unngå akkurat det på, unntatt å faktisk finne koden i FEST da selvfølgelig.

NamingSystem? :-)

**Comment by rockphotog** [comment]
Så, litt fart på forslaget - NamingSystem eller CodingSystem? @stinejohans1 og dere andre

`http://legemiddelverket.no/fhir/NamingSystem/fest-id`
`http://legemiddelverket.no/fhir/CodeSystem/fest-id`

**Comment by oaassv** [comment]
Beste praksis for tiden er vel at selve kodene uansett ikke bør ha semantisk verdi, men det er også et spørsmål om hva det refereres til - semantiske selvstendige entiteter (diagnosekoder) eller opplisting av populasjoner (ex fødselsnummer for innbyggere). Enig i at legemidler kanskje ligger i en gråsone her. 

Skal vi ta den under eventuelt i TSK i dag og få tatt en beslutning?

**Comment by rockphotog** [comment]
Ja, kan i hvert fall ta det opp - det er vel kanskje en litt halvfilosofisk sak. Som CodeSystem vil det nok være tomt, men det er jo teknisk mulig for f.eks. SLV å populere det (validering/oppslag av en kodeverdi skjer kanskje mot et ValueSet, for å tenke anvendelse).

**Comment by stinejohans1** [comment]
I SAFEST så har vi sett på FEST-ID'er som business id, og lagt det som identifier. Se https://simplifier.net/guide/SAFESTImplementationGuide/Identifiers

Mulig dette ikke blir helt riktig, men vi har her definert system med utgangspunkt i slik det er i basis-profilen til medication.

**Comment by rockphotog** [comment]
Joda, så spm er om man skal dele den opp i flere CodeSystem, som det ser ut som dere har tenkt med
` http://ehelse.no/fhir/CodeSystem/FEST/KatLegemiddelMerkevare`

**Comment by stinejohans1** [comment]
Ja, jeg anså det som et behov da vi på MedicinalProductManufactured trenger å referere til to forskjellige kataloger i FEST, legemiddeldose og legemiddelmerkevare.

**Comment by rockphotog** [comment]
@daghammer @olemartinwinnem Ka dokke tro?

**Comment by rockphotog** [comment]
Note to self: #29

**Comment by rockphotog** [comment]
Oppfølging fra TSK-møte 25.9.2020: @rockphotog samler SLV, SFM og E-helse i et rom, der vi bestemmer oss for endelige CodeSystem/NamingSystem-URI'er for de forskjellige FEST-id'ene det er behov for.

**Comment by oaassv** [comment]
@rockphotog Hva er status på samling i felles rom?

**Comment by rockphotog** [comment]
> Hva er status på samling i felles rom?

@oaassv På stedet hvil. Jeg har ikke kalt inn til møte med SLV og SFM som jeg planla - det får bli i 2021. Tenker at vi lager er forslag først.

**Comment by rockphotog** [comment]
Brukt i LMDI, midlertidig:

```
http://dmp.no/fhir/NamingSystem/festLegemiddelDose
http://dmp.no/fhir/NamingSystem/festLegemiddelMerkevare
http://dmp.no/fhir/NamingSystem/festLegemiddelPakning
http://dmp.no/fhir/NamingSystem/festLegemiddelVirkestoff
```

**Comment by rockphotog** [comment]
Vi må snart ta en dugnad og få landet denne - @thomiz @oyvindaa1 @HL7Norway/tsk

**Comment by oyvindaa1** [comment]
Her er det kanskje allerede tatt et valg i SAFEST?

**Comment by rockphotog** [comment]
Jeg lagde et forslag til SAFEST for veldig lenge siden, må uansett finne ut hva som er i bruk p.t.. 
eResept/SFM har uansett en helt annen filosofi.
